src/sentry/api/fields/sentry_slug.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from django.utils.translation import gettext_lazy as _
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import extend_schema_field
+from rest_framework import serializers
+
+# (?![0-9]+$) - Negative lookahead to ensure the slug is not entirely numeric
+# [a-z0-9_\-] - Matches lowercase letters, numbers, underscores, and hyphens
+MIXED_SLUG_PATTERN = r"^(?![0-9]+$)[a-z0-9_\-]+$"
+DEFAULT_SLUG_ERROR_MESSAGE = _(
+    "Enter a valid slug consisting of lowercase letters, numbers, underscores or hyphens. "
+    "It cannot be entirely numeric."
+)
+
+
+@extend_schema_field(field=OpenApiTypes.STR)
+class SentrySlugField(serializers.RegexField):
+    """
+    A regex field which validates that the input is a valid slug. Allowed
+    characters are lowercase letters, numbers, underscores, and hyphens. The
+    slug cannot be entirely numeric.
+    """
+
+    default_error_messages = {
+        "invalid": DEFAULT_SLUG_ERROR_MESSAGE,
+    }
+
+    def __init__(
+        self,
+        pattern=MIXED_SLUG_PATTERN,
```

Comment:
We let `pattern` be set because org slugs have a different pattern than other slugs (org slugs cannot contain hyphens or start with a dash)

Reply:
This looks dangerous because someone can set a pattern that results in numeric slug. Right? If it's just org vs. non-org can you generate the patterns here and the callsite only choose between the 2?